### DISCUSSION TITLE
asdf plugin for hledger-flow

### DISCUSSION THREAD

**Discussion by airtonix** [apauley/hledger-flow#118] [Show and tell]
If you're going to version control your plain text accounting projects, then you should also describe the version of the tools your projects use.

enter [asdf](https://asdf-vm.com/) and the thing I made: https://github.com/airtonix/asdf-hledger-flow.

I've made some plugins for asdf that download,install and manage the versions of hledger-flow your project describes.

First install [asdf](https://asdf-vm.com/)

then:
 
```sh
$ asdf plugin add hledger-flow https://github.com/airtonix/asdf-hledger-flow`

# installs the plugin

$ asdf list all hledger-flow
0.10-beta
0.11-beta
0.11.1.0-beta
0.11.1.1-beta
0.11.1.2-beta-RC1
0.11.1.2-beta-pre
0.11.1.2-beta
0.11.2.0-beta
0.11.3.0-beta
0.12.0.0-beta
0.12.1.0
0.12.2.0
0.12.2.1
0.12.3.0
0.12.3.1
0.12.4.0
0.13.0.0
0.13.1.0
0.13.2.0
0.14.0.0
0.14.1.0
0.14.2.0
0.14.3.0
0.14.4
0.15.0

$ asdf install hledger-flow 0.15.0

# downloads and installs v0.15.0 

$ asdf local hledger-flow 0.15.0

# creates ./.tool-versions with the contents: 
# hledger-flow 0.15.0
```

if you have other projects that rely on different versions of hledger-flow, do the same in those directories and when switching between directories a userland shim of `~/.asdf/shims/hledger-flow` automatically redirects execution flow to the right version of the binary based on the nearest found `.tool-versions` file.

**Comment by apauley** [comment]
This is great, thanks for adding it!
